src/modules/claims/claims.service.ts

```diff
@@ -1540,22 +1533,35 @@ export class ClaimsService {
    * Set the Verifier for Claim Issuance.
    *
    */
-  private _setClaimIssuerVerifier() {
+  private _setIssuerVerifier() {
     this._credentialResolver = new IpfsCredentialResolver(
       this._signerService.provider,
       this._didRegistry.registrySettings,
       this._didRegistry.ipfsStore
     );
     const domainReader = this._domainsService.domainReader;
     this._issuerResolver = new EthersProviderIssuerResolver(domainReader);
-    this._vcIssuerVerifier = new VCIssuerVerification(
+    const revokerResolver = new EthersProviderRevokerResolver(domainReader);
+    const verifyProof = (vc: string, proofOptions: string) =>
+      this._verifiableCredentialService.verify(
+        JSON.parse(vc),
+        JSON.parse(proofOptions)
+      );
+    const revocationVerification = new RevocationVerification(
+      revokerResolver,
       this._issuerResolver,
       this._credentialResolver,
-      (vc: string, proofOptions: string) =>
-        this._verifiableCredentialService.verify(
-          JSON.parse(vc),
-          JSON.parse(proofOptions)
-        )
+      this._signerService.provider,
+      this._didRegistry.registrySettings,
+      verifyProof
+    );
+    this._issuerVerification = new IssuerVerification(
+      this._issuerResolver,
+      this._credentialResolver,
+      this._signerService.provider,
+      this._didRegistry.registrySettings,
+      revocationVerification,
+      verifyProof
```

Comment:
Do we need to `bind(this)` here so that `this._verifiableCredentialService.verify` in the `verifyProof` implementation works as expected?
```suggestion
      verifyProof.bind(this)
```

Reply:
Hmm, I am not sure why. `this` is already set based on the scope `verifyProof` is defined with, so it will be `ClaimsService`

Reply:
My thinking is that `this` changes in the context of `IssuerVerification`. But actually, I think you're right, because `verifyProof` is an arrow function, I think it captures the value of `this` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions 

> However, because it is a [closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures), not the function's own binding, the value of this will not change based on the execution context.